app/models/user_extension/groups.rb

```diff
@@ -46,10 +46,18 @@ def committees
         def councils
           self.select{|group|group.council?}
         end
+        def by_visited
+          self.order('memberships.visited_at DESC')
+        end
+        # groups we have visited most recently, including their parent groups.
         def recently_active(options={})
-          options[:limit] ||= 13
-          limit(options[:limit]).order('memberships.visited_at DESC')
-            .where('groups.type IS NULL').all
+          options[:limit] ||= 20
+          grps = []
+          self.by_visited.limit(options[:limit]).all.each do |group|
+            grps << group
+            grps << group.parent if group.parent
+          end
+          grps.sort_by{|g|g.name}.uniq{|g|g.name}
```

Comment:
> not possible, since the list needs to ensure that the committees come after the parent group.

to clarify, the committees are not included in the list. i thought this was a good idea, since i often work mostly in committees and i was always annoyed they don't show up.

but, to make the committees not look weird, they are nested (indented) under the parent group. in order to get this tree structure to display right, the sorting needs to ensure that a group's committees all come directly after the group in the list. i suppose you could achieve this in other ways, but shorting by name is sure the easiest.